redbot/core/utils/menus.py

```diff
@@ -1,29 +1,1262 @@
+                else:
+                    yield handler
+
+    async def __call_handlers(
+        self, args, event: ReactionEvent, emoji: Optional[str] = None
+    ) -> None:
+        if emoji not in self._HANDLERS and None not in self._HANDLERS:
+            return
+        await asyncio.gather(*(handler(*args) for handler in self.__iter_handlers(event, emoji)))
+
+
+class PagedMenu(ReactionMenu, exit_button=True, initial_emojis=("⬅", "❌", "➡")):
+    """A reaction menu for scrolling through pages.
 
+    By default, this menu's intial reactions will be ⬅ ❌ ➡, with the
+    cross being the exit button.
 
+    Below are the keyword arguments you must/can pass to the
+    :meth:`send_and_return` or :meth:`send_and_wait` classmethods.
+
+    .. versionadded:: 3.2
+
+    Keyword Arguments
+    -----------------
+    pages : Iterable[Union[str, discord.Embed, Tuple[str, discord.Embed]]], **Required**
+        An iterable of pages which the menu can scroll through. It can
+        be a combination of strings (for text content), embeds, or
+        2-tuples containing (text, embed).
+    pagenum_in_footer: bool
+        Any embeds which don't already have non-empty footer text will
+        have the following added to it: *Page <k>/<n>*, where *<k>* is
+        the current page number and *<n>* is the number of pages.
+        Defaults to ``True``.
+    footer_text : Optional[str]
+        Text to add to the footer of any embeds which don't already have
+        non-empty footer text. If ``pagenum_in_footer`` is ``True``,
+        this will be added like so: *Page <k>/<n> | <footer_text>*.
+    first_page : int
+        What the initial page index should be. Defaults to zero.
+    arrows_always : bool
+        When ``True``, the arrow reactions for previous and next page
+        buttons will be added, even when there is only one page in
+        ``pages``. This is only really useful for subclasses who might
+        only start with one page, but generate the rest dynamically.
+        Defaults to ``False``.
+
+    Attributes
+    ----------
+    _pages : List[Union[str, discord.Embed, Tuple[str, discord.Embed]]]
+        The list of pages provided. This is documented for the
+        interest of subclasses only.
+    _cur_page : int
+        The current page number, indexed from zero. This should always
+        be in the range [0, ``len(_pages)``). This is documented for the
+        interest of subclasses only.
+
+    """
+
+    _pages: List[_Page]
+    _cur_page: int
+
+    def __init__(
+        self,
+        *,
+        pages: Iterable[_Page],
+        footer_text: Optional[str] = None,
+        first_page: int = 0,
+        arrows_always: bool = False,
+        **kwargs,
+    ) -> None:
+        self._pages = list(pages)
+        self._cur_page = first_page
+        self._footer_text = footer_text
+
+        if arrows_always is False and len(self._pages) == 1:
+            initial_emojis = ["❌"]
+        else:
```

Comment:
logic here doesnt appear to pass the eye test. the if doesnt check that there arent emojis in constructer. this should determine the default. This should also probably use kwargs.pop with the use of super below

Reply:
Thanks for pointing this out - should be fixed up now.